### PR TITLE
Remove services and topics from BrakeTestExecutor

### PR DESCRIPTION
**Pull request by hslusarek**

### CODE REVIEW

File: prbt_hardware_support/src/brake_test_executor.cpp [hunk 1]
```diff
@@ -15,92 +15,85 @@
...
   }
-
-  BrakeTest braketest_srv;
-  if (!trigger_braketest_client_.call(braketest_srv))
+  else
   {
-    response.success = false;
-    response.error_msg = "Failed to trigger brake test via service " + trigger_braketest_client_.getService();
-    response.error_code.value = BrakeTestErrorCodes::TRIGGER_BRAKETEST_SERVICE_FAILURE;
+    ROS_WARN_STREAM("Calling hold on controller skipped because no hold service given");
```
Comment (agutenkunst) [inline prbt_hardware_support/src/brake_test_executor.cpp]: We need better error handling. If a hold is not possible the whole system is in a critical state...

Comment (agutenkunst) [reply prbt_hardware_support/src/brake_test_executor.cpp]: There is always true returned...

File: prbt_hardware_support/src/brake_test_executor_node.cpp [hunk 2]
```diff
@@ -26,7 +44,28 @@ int main(int argc, char **argv)
   ros::init(argc, argv, "brake_test_executor");
   ros::NodeHandle nh{"~"};
 
-  prbt_hardware_support::BrakeTestExecutor brake_test_executor(nh);
+  waitForService(CONTROLLER_HOLD_MODE_SERVICE_NAME);
```
Comment (agutenkunst) [inline prbt_hardware_support/src/brake_test_executor_node.cpp:47]: What if the service is missing ?

File: prbt_hardware_support/src/brake_test_executor_node.cpp [hunk 3]
```diff
@@ -26,7 +44,28 @@ int main(int argc, char **argv)
...
+  ros::ServiceClient hold_client = nh.serviceClient<std_srvs::Trigger>(CONTROLLER_HOLD_MODE_SERVICE_NAME);
+
+  waitForService(BRAKETEST_ADAPTER_SERVICE_NAME);
```
Comment (agutenkunst) [inline prbt_hardware_support/src/brake_test_executor_node.cpp:50]: What if the service is missing ?

File: prbt_hardware_support/src/brake_test_executor_node.cpp [hunk 4]
```diff
@@ -26,7 +44,28 @@ int main(int argc, char **argv)
...
+  ros::ServiceClient brake_test_execute_client = nh.serviceClient<BrakeTest>(BRAKETEST_ADAPTER_SERVICE_NAME);
+
+  waitForService(CONTROLLER_UNHOLD_MODE_SERVICE_NAME);
```
Comment (agutenkunst) [inline prbt_hardware_support/src/brake_test_executor_node.cpp:53]: What if the service is missing ?

File: prbt_hardware_support/src/brake_test_executor_node.cpp [hunk 5]
```diff
@@ -26,7 +44,28 @@ int main(int argc, char **argv)
...
+  ros::ServiceClient unhold_client = nh.serviceClient<std_srvs::Trigger>(CONTROLLER_UNHOLD_MODE_SERVICE_NAME);
+
+  waitForService(BRAKE_TEST_RESULT_SERVICE_NAME);
```
Comment (agutenkunst) [inline prbt_hardware_support/src/brake_test_executor_node.cpp:56]: What if the service is missing ?

**Review by agutenkunst — CHANGES_REQUESTED** [review]
There are some questions but I can see that these were also relevant before this PR.

File: prbt_hardware_support/src/brake_test_executor.cpp (hunk 1 above)
Comment (jschleicher) [reply prbt_hardware_support/src/brake_test_executor.cpp]: And this DOES change behaviour (hold was required before).

Comment (jschleicher) [reply prbt_hardware_support/src/brake_test_executor.cpp]: > There is always true returned...

Services always return true, if the result type was filled, see [ROS wiki](http://wiki.ros.org/roscpp/Overview/Services#Callback_Signature).

Comment (martiniil) [reply prbt_hardware_support/src/brake_test_executor.cpp]: Yes, I agree. Can we do the error handling separately?

File: prbt_hardware_support/src/brake_test_executor_node.cpp (hunk 2 above)
Comment (martiniil) [reply prbt_hardware_support/src/brake_test_executor_node.cpp:47]: waits infinitely

File: prbt_hardware_support/src/brake_test_executor_node.cpp (hunk 3 above)
Comment (martiniil) [reply prbt_hardware_support/src/brake_test_executor_node.cpp:50]: waits infinitely

File: prbt_hardware_support/src/brake_test_executor_node.cpp (hunk 4 above)
Comment (martiniil) [reply prbt_hardware_support/src/brake_test_executor_node.cpp:53]: waits infinitely

File: prbt_hardware_support/src/brake_test_executor_node.cpp (hunk 5 above)
Comment (martiniil) [reply prbt_hardware_support/src/brake_test_executor_node.cpp:56]: waits infinitely

File: prbt_hardware_support/src/brake_test_executor.cpp (hunk 1 above)
Comment (martiniil) [reply prbt_hardware_support/src/brake_test_executor.cpp]: This PR should not change behaviour. I will correct it.

File: prbt_hardware_support/src/brake_test_executor_node.cpp (hunk 2 above)
Comment (agutenkunst) [reply prbt_hardware_support/src/brake_test_executor_node.cpp]: I don't think so...

Comment (martiniil) [reply prbt_hardware_support/src/brake_test_executor_node.cpp:47]: waits until ros shutdown

**Pull request merged**